python/foglamp/services/core/service_registry/monitor.py

```diff
@@ -63,9 +71,22 @@ def __init__(self):
             for service_record in ServiceRegistry.all():
                 if service_record._id not in check_count:
                     check_count.update({service_record._id: 1})
+
                 # Try ping if service status is either running or doubtful (i.e. give service a chance to recover)
-                if service_record._status not in [ServiceRecord.Status.Running, ServiceRecord.Status.Unresponsive]:
+                if service_record._status not in [ServiceRecord.Status.Running,
+                                                  ServiceRecord.Status.Unresponsive,
+                                                  ServiceRecord.Status.Failed]:
+                    continue
+
+                self._logger.debug("Service: {} Status: {}".format(service_record._name, service_record._status))
+
+                if service_record._status == ServiceRecord.Status.Failed:
+                    if self._restart_failed == "auto":
+                        if service_record._id not in self.restarted_services:
```

Comment:
as we have `self.restarted_services.remove(service_record._id)` once restarted, it looks good to me.